### PR TITLE
Press more juice

### PR DESCRIPTION
**Pull request by pyricau**
Remove extra waste of string comparison when going through bind class hierarchy

### CODE REVIEW

**Comment by pyricau** [comment]
This only improves the situation if you get several miss that all go up until the framework class.

**Comment by JakeWharton** [comment]
Test needs updated

**Comment by pyricau** [comment]
Updated, tests fixed.

File: butterknife/src/test/java/butterknife/ButterKnifeTest.java [hunk 1]
```diff
@@ -138,11 +138,11 @@ public void resetViewsCache() {
     assertThat(ButterKnife.BINDERS).contains(entry(Example.class, ButterKnife.NOP_VIEW_BINDER));
   }
 
-  @Test public void bindingKnownPackagesIsNoOp() {
+  @Test public void bindingKnownPackagesCaches() {
     ButterKnife.bind(new Activity());
-    assertThat(ButterKnife.BINDERS).isEmpty();
+    assertThat(ButterKnife.BINDERS).hasSize(1);
```
Comment (JakeWharton) [inline butterknife/src/test/java/butterknife/ButterKnifeTest.java]: This needs to assert the no-op binding was the one which was cached.

**Comment by JakeWharton** [comment]
Fixed as e9c1533e2af2b26bf26ce13c139ae6ffdeb3a7c5

**Pull request closed without merging**